attribution-data-file-share/src/main/java/gov/cms/ab2d/attributionDataShare/AttributionDataShareHandler.java

```diff
@@ -0,0 +1,87 @@
+package gov.cms.ab2d.attributionDataShare;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.LambdaLogger;
+import com.amazonaws.services.lambda.runtime.RequestStreamHandler;
+import gov.cms.ab2d.lambdalibs.lib.PropertiesUtil;
+
+import java.io.*;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Properties;
+
+public class AttributionDataShareHandler implements RequestStreamHandler {
+
+    private static final String FILE_PATH = "/opt/";
+    private static final String FILE_PARTIAL_NAME = "ab2d-beneids_";
+    private static final String FILE_FORMAT = ".txt";
+
+    private static final String SELECT_ALL_FROM_COVERAGE = "SELECT DISTINCT current_mbi FROM public.coverage";
+
+    // Writes out a file to the FILE_PATH.
+    // I.E: "ab2d-beneids_2023-08-16T12:08:56.235-0700.txt"
+    public void handleRequest(InputStream inputStream, OutputStream outputStream, Context context) throws IOException {
+        LambdaLogger logger = context.getLogger();
+        logger.log("AttributionDataShare Lambda is started");
+
+        String currentDate = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ").format(new Date());
+        String fileFullPath = FILE_PATH + FILE_PARTIAL_NAME + currentDate + FILE_FORMAT;
+
+        try (FileWriter fileWriter = new FileWriter(fileFullPath)) {
+            Connection dbConnection = getConnection();
+
+            Statement statement = dbConnection.createStatement();
+            ResultSet resultSet = statement.executeQuery(SELECT_ALL_FROM_COVERAGE);
+
+            BufferedWriter bufferedWriter = new BufferedWriter(fileWriter);
+
+            if (resultSet.next() == false) {
+                logger.log("ResultSet is empty...skipping writing to file");
+            } else {
+                logger.log("Writing to file...");
+                do {
+                    String result = resultSet.getString(1);
+                    if (result != null || result != "") {
+                        bufferedWriter.write(result);
+                    }
+                } while (resultSet.next());
```

Comment:
When you combined query processing and writing into 1 loop, you reduced the space complexity because you are not using List<String>, but the execution time will not change significantly because the main thread will wait for the selection results and only then write to the file.
I think we should make queries and writes asynchronous.

Reply:
Okay, I thought by reducing from 2 loops to one it would increase performance. Do I need to go back to the way it was before? I am unsure how to resolve this. Does java have a way to asynchronously stream results of a query from the db?

Reply:
`Do I need to go back to the way it was before?` - No, it's better now, but there is a room to increase more

`Does java have a way to asynchronously stream results of a query from the db?` - I'm afraid, I don't know such library. It should be like... when select returns some part of the ids, you write it to a file and simultaneously execute select on the next chunk.

Reply:
https://stackoverflow.com/questions/4087696/is-asynchronous-jdbc-call-possible

This is from a long time ago (12 years old). But from my own searching I cannot find an easy implementation to make the database query asynchronous. 

I am wondering if using the scrollable resultset would be enough? 

Unless we make the file write asyncronos. This might be possible? Thoughts?

Reply:
Let's see what we will have after load testing.
We can write async to file, but I think this is what we can do in follow up ticket for load testing